[docs] clarify logsSubscribe `mentions` param (#31225)
solana-labs/solana: docs/src/api/websocket/_logsSubscribe.mdx

A string with one of the following values:

- `all` - subscribe to all transactions except for simple vote transactions

> - `allWithVotes` - subscribe to all transactions including simple vote

Reviewer: ```suggestion
- `allWithVotes` - subscribe to all transactions, including simple vote
```

  transactions

</Field>

<Field type="object">

An object with the following field:

> - `mentions: [ <string> ]` - array containing a single Pubkey (as base-58
>   encoded string) to listen for being mentioned in any transaction

Reviewer: ```suggestion
- `mentions: [ <string> ]` - array containing a single Pubkey (as base-58
  encoded string); if present, subscribe to only transactions mentioning this address
```
